new/rfcXXX-distro-sugar-helpers.md

```diff
@@ -0,0 +1,159 @@
+unmodified, along with the `systemd?`, `docker?` and `kitchen?` helpers.  Since they have been
+in widespread use, it is important to not modify their meaning.
+
+### Architecture Helpers
+
+* `_64_bit?`
+* `_32_bit?`
+* `i386?`
+* `intel?`
+* `sparc?`
+* `ppc64?`
+* `ppc64le?`
+* `powerpc?`
+* `armhf`?
+* `s390x`?
+
+### Platform Helpers
+
+We will not be pulling in the `debian_before_squeeze?` style of `platform` and `platform_version` helpers
+at this time (simply limiting the scope, not a statement that we do or do not eventually want them).
+
+The complete list of platform helpers is long, examples:
+
+* `amazon_linux?`
+* `centos?`
+* `linux_mint?`
+* `oracle_linux?`
+* `redhat_enterprise_linux?`
+* `scientific_linux?`
+* `ubuntu?`
+* `solaris2?`
+* `aix?`
+* `smartos?`
+* `omnios?`
+* `raspbian?`
+* `nexus?`
+* `ios_xr?`
+
+Note that this list deviates from the platform definitions in core chef, to eliminate ambiguity we will also
+add helpers which precisely match the field in core chef:
+
+* `linuxmint?`
+* `oracle?`
+* `scientific?`
+* `redhat?`
+
+Additional helpers will also be added to fill out the complete list of platforms, which will match the fields
+used by core chef (matching the code in ohai).
+
+Where there is a collision between `platform` and `platform_family` being named the same thing, and the platform_family
+contains multiple distros, the platform version will be renamed:
+
+* `debian_platform?`
+* `fedora_platform?`
+
+This is driven mostly by the widespread adoption of the `debian?` helper to mean "debian platform family".
+
+For distros like "freebsd" where the platform, platform_family and os all match and there are no other derived classes
+we shall use `freebsd?` to refer to the platform.
+
+### Platform Family Helpers
+
+Similarly we include the existing chef-sugar helpers:
+
+* `arch_linux?`
+* `debian?`
+* `fedora?`
+* `freebsd?`
+* `gentoo?`
+* `linux?`
+* `mac_os_x?`
+* `openbsd?`
+* `rhel?`
+* `slackware?`
+* `suse?`
+* `windows?`
+* `wrlinux?`
+
+Where the chef-sugar helper is different from the name of the `platform_family` the name from ohai will also be added and
+the full list from ohai data will be filled out.
+
+### Additional Helpers From Chef-Sugar
+
+These three helper will also be included:
+
+* `docker?`
+* `systemd?`
+* `kitchen?`
+
+The `docker?` helper has already been merged as of Chef 12.11.18
+
+### Higher Level Helpers
+
+In addition to the chef-sugar helpers, higher level helpers will be added:
+
+* `rpm_based?` -- this will include the `platform_family` of `rhel`, `fedora`, `amazon`, and `suse`
+* `fedora_based?` -- this will include the `platform_family` of `rhel`, `fedora`, and `amazon`
+* `solaris_based?` -- this will include the `platform` of `openindiana`, `opensolaris`, `nexentacore`, `omnios`, `solaris2`, and `smartos`
+* `bsd_based?` -- this will include the `platform` of `freebsd`, `netbsd`, `openbsd`, and `dragonflybsd`
```

Comment:
yeah.  i was trying to get at that with the latter statement.  IDK exactly how to write that right now since i'm coming down with a cold.